src/core/datastore/sync/sync-manager.js

```diff
@@ -60,8 +62,31 @@ export class SyncManager {
     if (isEmpty(collection)) {
       return Promise.reject(new KinveyError('Invalid or missing collection name'));
     }
-    return this._fetchItemsFromServer(collection, query, options)
-      .then(entities => this._replaceOfflineEntities(collection, query, entities));
+
+    return this.getSyncItemCountByEntityQuery(collection, query)
```

Comment:
@georgiwe Not sure if this is being done somewhere else but please verify that we are not duplicating this check. I had to add this here to fix a failing unit tests. All unit tests pass now and I wasn't able to find this check anywhere else after doing a search.

Reply:
Which test, why was it failing? The reason this was in CacheStore.pull() was that it's logic that isn't part of `pull()`. I like to keep concerns separate. `SyncManager.pull()` does work, CacheStore.pull() should make sure "business logic" like this is handled before calling it to do the work.

Reply:
I was referring to [this unit test](https://github.com/Kinvey/js-sdk/pull/247/files#diff-50ea438538b5fab8b06c67a1ba2639caR230)

Reply:
I am seeing some weird behavior that is causing the unit tests [here](https://github.com/Kinvey/js-sdk/pull/247/files#diff-50ea438538b5fab8b06c67a1ba2639caR114) and [here](https://github.com/Kinvey/js-sdk/pull/247/files#diff-50ea438538b5fab8b06c67a1ba2639caR230) to fail however I have debugged and the logic appears to be correct. Maybe another pair of eyes on the unit tests will help see something I am missing.